plugin/storage/cassandra/schema/create.go

```diff
@@ -0,0 +1,216 @@
+		if strings.TrimSpace(line) != "" {
+			filteredLines = append(filteredLines, line)
+		}
+	}
+
+	cqlOutput = strings.Join(filteredLines, "\n")
+
+	replacements := map[string]string{
+		"${keyspace}":               params.Keyspace,
+		"${replication}":            params.Replication,
+		"${trace_ttl}":              fmt.Sprintf("%d", params.TraceTTL),
+		"${dependencies_ttl}":       fmt.Sprintf("%d", params.DependenciesTTL),
+		"${compaction_window_size}": fmt.Sprintf("%d", params.CompactionWindowSize),
+		"${compaction_window_unit}": params.CompactionWindowUnit,
+	}
+
+	for placeholder, value := range replacements {
+		cqlOutput = strings.ReplaceAll(cqlOutput, placeholder, value)
+	}
+
+	return cqlOutput, nil
+}
+
+func getEnv(key, defaultValue string) string {
+	if value, exists := os.LookupEnv(key); exists {
+		return value
+	}
+	return defaultValue
+}
+
+func isValidKeyspace(keyspace string) bool {
+	match, _ := regexp.MatchString(`^[a-zA-Z0-9_]+$`, keyspace)
+	return match
+}
+
+func getCompactionWindow(traceTTL int, compactionWindow string) (int, string, error) {
+	var compactionWindowSize int
+	var compactionWindowUnit string
+
+	if compactionWindow != "" {
+		matched, err := regexp.MatchString(`^[0-9]+[mhd]$`, compactionWindow)
+		if err != nil {
+			return 0, "", err
+		}
+
+		if matched {
+			numStr := strings.TrimRight(compactionWindow, "mhd")
+			unitStr := strings.TrimLeft(compactionWindow, numStr)
+
+			compactionWindowSize, err = strconv.Atoi(numStr)
+			if err != nil {
+				return 0, "", errors.New("invalid compaction window size format")
+			}
+			compactionWindowUnit = unitStr
+		} else {
+			return 0, "", errors.New("invalid compaction window size format. Please use numeric value followed by 'm' for minutes, 'h' for hours, or 'd' for days")
+		}
+	} else {
+		traceTTLMinutes := traceTTL / 60
+		compactionWindowSize = (traceTTLMinutes + 30 - 1) / 30
+		compactionWindowUnit = "m"
+	}
+
+	switch compactionWindowUnit {
+	case "m":
+		compactionWindowUnit = "MINUTES"
+	case "h":
+		compactionWindowUnit = "HOURS"
+	case "d":
+		compactionWindowUnit = "DAYS"
+	default:
+		return 0, "", errors.New("invalid compaction window unit")
+	}
+
+	return compactionWindowSize, compactionWindowUnit, nil
+}
+
+func (mb *MappingBuilder) GetSpanServiceMappings() (spanMapping string, serviceMapping string, err error) {
+	traceTTL, _ := strconv.Atoi(getEnv("TRACE_TTL", "172800"))
+	dependenciesTTL, _ := strconv.Atoi(getEnv("DEPENDENCIES_TTL", "0"))
+	cas_version := getEnv("VERSION", "4")
+	// template := os.Args[1]
+	var template string
+	var cqlOutput string
+	if template == "" {
+		switch cas_version {
+		case "3":
+			template = "./v003.cql.tmpl"
+		case "4":
+			template = "./v004.cql.tmpl"
+		default:
+			template = "./v004.cql.tmpl"
+		}
+		mode := getEnv("MODE", "")
+
+		if mode == "" {
+			fmt.Println("missing MODE parameter")
+			return
+		}
+		var datacentre, replications string
+		// var ReplicationFactor int
+		if mode == "prod" {
```

Comment:
I don't think we need this distinction anymore. Users can provide precise values for parameters, we don't have to have Jaeger guess those parameters based on the MODE

Reply:
so should i remove the mode attribute and take prod mode as default 
?

Reply:
yes. Whichever parameters the prod attribute affects today should simply be exposed as configuration options for the user.

Reply:
btw, we do not need to enable this schema-management capability in jaeger-v1, especially if we need new configuration for it - we can make it only configurable in v2 (which is easier, just define the struct with `mapstructure` tags)